ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -625,6 +615,129 @@ public Boolean get() {
     }
   }
 
+  private static final class PullMetricsInitializer {
+
+    private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
+    private final SlidingWindowRateLimiter pullBandRateLimiter;
+
+    private PullMetricsInitializer(
+        final Optional<PullQueryExecutorMetrics> pullQueryMetrics,
+        final SlidingWindowRateLimiter pullBandRateLimiter) {
+
+      this.pullQueryMetrics = Objects.requireNonNull(pullQueryMetrics, "pullQueryMetrics");
+      this.pullBandRateLimiter = Objects.requireNonNull(pullBandRateLimiter, "pullBandRateLimiter");
+    }
+
+    private MetricsCallback initializeTableMetricsCallback(
+        final AtomicReference<PullQueryResult> resultForMetrics) {
```

Comment:
Thanks for factoring these out from the method above. I'm wondering if we can skip this extra class definition, though. It seems like just having a static method would do just fine and would save on both lines of code and on complexity for us to understand the code.

```suggestion
  private static MetricsCallback initializeTableMetricsCallback(
      final Optional<PullQueryExecutorMetrics> pullQueryMetrics,
      final SlidingWindowRateLimiter pullBandRateLimiter
      final AtomicReference<PullQueryResult> resultForMetrics) {
```

(for example... I'd propose to make the other method also just a static method by adding these two arguments)